projects/packages/sync/src/class-dedicated-sender.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Dedicated Sender.
+ *
+ * The class is responsible for spawning dedicated Sync requests.
+ *
+ * @package automattic/jetpack-sync
+ */
+
+namespace Automattic\Jetpack\Sync;
+
+use WP_Error;
+/**
+ * Class to manage Sync spawning.
+ * The purpose of this class is to provide the means to unblock Sync from running in the shutdown hook
+ * of regular requests by spawning a dedicated Sync request instead which will trigger Sync to run.
+ */
+class Dedicated_Sender {
+
+	/**
+	 * Check if this request should trigger Sync to run.
+	 *
+	 * @access public
+	 *
+	 * @param Automattic\Jetpack\Sync\Queue $queue Queue object.
+	 *
+	 * @return boolean|WP_Error True if this is a POST request and jetpack_dedicated_sync_request is set, false otherwise, WP_Error if the nonce can't be verified.
+	 */
+	public static function is_dedicated_sync_request( $queue ) {
+
+		$is_dedicated_sync_request = isset( $_SERVER['REQUEST_METHOD'] ) &&
+			'POST' === $_SERVER['REQUEST_METHOD'] &&
+			isset( $_POST['jetpack_dedicated_sync_request'] );
+
+		if ( $is_dedicated_sync_request ) {
+			$is_valid = isset( $_POST['nonce'] ) &&
```

Comment:
what is the benefit of a nonce check on this request?

Reply:
Removed in https://github.com/Automattic/jetpack/pull/22506/commits/5a898225746f094471ec139fc6d8375f8e8a404c